addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1063,6 +1239,65 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
     _getSelect: function () {
         return this.$target[0].querySelector('select');
     },
+    /**
+     * Fills the we-select with existing fields in the form
+     */
+    async _updateAvailableVisibilityDependencies() {
+        const selectEl = document.querySelector('we-select[data-name="hidden_condition_opt"]');
+        while (selectEl.firstChild) {
+            selectEl.removeChild(selectEl.lastChild);
+        }
```

Comment:
Probably means something is wrong, better to rerender the whole UI (like it seems already done via this.rerender)